addons/mail/tests/common.py

```diff
@@ -528,19 +543,29 @@ def _find_mail_mail_wpartners(self, recipients, status, mail_message=None, autho
             )
         return mail
 
-    def _find_mail_mail_wemail(self, email_to, status, mail_message=None, author=None, content=None, email_from=None):
+    def _find_mail_mail_wemails(self, email_to_all, status, mail_message=None, author=None, content=None, email_from=None):
         """ Find a mail.mail record based on various parameters, notably a list
         of email to (string emails).
 
-        :param email_to: either matching mail.email_to value, either a mail sent
-          to a single recipient whose email is email_to;
+        :param email_to_all: comma-separated or python list of email addresses comprising either
+          all email_to addresses of the message, all emails of the recipients or all emails of both;
 
         :return mail: a ``mail.mail`` record generated during the mock and matching
           given parameters and filters;
         """
         filtered = self._filter_mail(status=status, mail_message=mail_message, author=author, content=content, email_from=email_from)
+
+        if isinstance(email_to_all, str):
+            email_to_all = email_split(email_to_all)
+        email_to = (
+            {email_normalize(email) or email for email in email_to_all or []} - {''}
+        ) or ([email_to_all] if email_to_all else set())
```

Comment:
Feels like it makes the tool a bit hard to understand / use. It was previously tailored for single email I think (or a formatted comma separated for email_to ? not sure if we have cases ?)

So maybe _find_mail_mail_wemail and _find_mail_mail_wemails could exist and just call the bulk check moved in a sub-tool ? That way we have 2 cleaner APIs.